### DISCUSSION TITLE
Announcement: IKOS 3.2

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [NASA-SW-VnV/ikos#257] [Announcements]
The IKOS team is delighted to announce the release of IKOS 3.2. This is a maintenance release that addresses bugs related to Python packaging, LLVM 14, IPv6 handling and multi OS support. Other changes focus on the documentation, the software engineering process and leveraging features offered by Github.

There are two notable changes:
- We've dropped support for Python 2.
- We've added CI jobs via Github actions, which should ensure that IKOS keeps working on supported platforms.

For a detailed list of changes, see: https://github.com/NASA-SW-VnV/ikos/releases/tag/v3.2

## Metrics

23 issues were closed as part of IKOS 3.2.

The net balance including all files (e.g., code, docs) is that IKOS 3.2 is 437 lines _smaller_ than the previous version.

We had submissions from 4 new authors who had never submitted patches to IKOS before. Thanks!

**Comment by ivanperez-keera** [comment]
Massive thanks to @arthaud for his invaluable help. This would not have been published if it wasn't for him.

**Comment by asensi** [comment]
Thanks for the release!

About reducing workload for the next time:

The maintenance and development burden of a project can be reduced by not periodically changing every copyright year. There is the example of Curl: 
    - https://daniel.haxx.se/blog/2023/01/08/copyright-without-years/
    - https://github.com/curl/curl/commit/2bc1d775f510196154283374284f98d3eae03544

That's also supported by Steve Winslow [^1] : https://www.linuxfoundation.org/blog/blog/copyright-notices-in-open-source-software-projects

[^1]: Steve Winslow runs the Linux Foundation’s license scanning and analysis service, advising projects about licenses identified in their source code and dependencies. Steve is also involved with projects including SPDX, FOSSology and the Community Data License Agreement; manages The Linux Foundation’s trademark program; and assists on other legal matters. Previously, Steve was Vice President of Technology Law at Intralinks and an associate at Choate, Hall and Stewart in Boston. Steve graduated from Georgetown University Law Center and majored in computer science at Williams College.
        -- https://events.linuxfoundation.org/mentorship-session-os-licensing/

**Comment by asensi** [comment]
"Every January 1st, an army of open source developers rushes out to update their copyright attributions in licenses and documentation. Why? Because we’ve always done it that way.

I’ve stopped participating after I learned that copyright statements need only the year of the first publication and no lawyer that I asked contradicted.

Now, I’m not a lawyer, so don’t take this as legal advice from me. All I’m saying is that if it’s good enough for Google’s, Microsoft’s, and Netflix’s lawyers, it’s good enough for me:

 - [Google’s Go](https://github.com/golang/go/blob/master/LICENSE)
 - [Microsoft’s VS Code](https://github.com/microsoft/vscode/blob/main/LICENSE.txt)
 - [Netflix’s Hystrix](https://github.com/Netflix/Hystrix/blob/master/README.md#license)

In fact, you can drop years altogether:

 - Linux Foundation’s [guidance](https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/).
 - Facebook [removed years](https://github.com/facebook/react/commit/b87aabdfe1b7461e7331abb3601d9e6bb27544bc) from React.
 - Amazon [switched positions](https://mastodon.social/@mark@chatter.quux.world/109621623369696598) in [2020](https://twitter.com/ajorg/status/1228369968963604480), too.

Enjoy your New Year doing something more fun!"
        -- Hynek Schlawack, https://hynek.me/til/copyright-years/

**Reply by ivanperez-keera** [reply]
Thanks, @asensi! I'm linking to these comments from the IKOS 3.3. planning discussion so that we keep it in mind for that release.
